src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
-
+/* eslint-disable */
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const historyChanges = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = {
+          ...newState,
+          ...action.extraData,
+        };
+        break;
+      case 'removeProperties':
+        for (const fieldName of action.keysToRemove) {
+          delete newState[fieldName];
+        }
+        break;
+      case 'clear':
```

Comment:
Adding an empty line after `break` in a `switch` is not mandatory, but it can improve code readability.

```suggestion
        break;
        
      case 'clear':
```